Dev

ahmadiq opened the pull request.

Commit:
init commit

Commit:
minor fix

Commit:
patch deployment with example container

Commit:
keycloak proxy container

Commit:
format

Commit:
fix patch

Commit:
temp changes for patch payload

Commit:
temp changes for patch payload

Commit:
cleanup

Commit:
add service update logic
add all currently used annotations

Commit:
fixes

Commit:
add checks for daemonset and statefulset

Commit:
revert version to 0.0.1

Commit:
use common config from own configmap instead of external keycloak configmap

Commit:
fixes

Commit:
fixes

stakater-user commented:
@ahmadiq Yikes! You better fix it before anyone else finds out! [Build 1](https://jenkins.tools.stackator.com/job/Stakater/job/ProxyInjector/job/PR-1/1/) has Failed!

Commit:
fix config runtime error
creating configmap with config.yml for proxy sidecar using proxyinjector config.yml

stakater-user commented:
@ahmadiq Yikes! You better fix it before anyone else finds out! [Build 2](https://jenkins.tools.stackator.com/job/Stakater/job/ProxyInjector/job/PR-1/2/) has Failed!

Commit:
dynamic annotations
handle config file

stakater-user commented:
@ahmadiq Yikes! You better fix it before anyone else finds out! [Build 3](https://jenkins.tools.stackator.com/job/Stakater/job/ProxyInjector/job/PR-1/3/) has Failed!

Commit:
proper handling of resource uri

stakater-user commented:
@ahmadiq Yikes! You better fix it before anyone else finds out! [Build 4](https://jenkins.tools.stackator.com/job/Stakater/job/ProxyInjector/job/PR-1/4/) has Failed!

Commit:
struct implementation for config

Commit:
use config as container args instead of configmap

stakater-user commented:
@ahmadiq Yikes! You better fix it before anyone else finds out! [Build 5](https://jenkins.tools.stackator.com/job/Stakater/job/ProxyInjector/job/PR-1/5/) has Failed!

Commit:
readme

stakater-user commented:
@ahmadiq Yikes! You better fix it before anyone else finds out! [Build 6](https://jenkins.tools.stackator.com/job/Stakater/job/ProxyInjector/job/PR-1/6/) has Failed!

Commit:
remove debug logging

stakater-user commented:
@ahmadiq Yikes! You better fix it before anyone else finds out! [Build 7](https://jenkins.tools.stackator.com/job/Stakater/job/ProxyInjector/job/PR-1/7/) has Failed!

stakater-user commented:
@ahmadiq Yikes! You better fix it before anyone else finds out! [Build 8](https://jenkins.tools.stackator.com/job/Stakater/job/ProxyInjector/job/PR-1/8/) has Failed!

Commit:
add pkg/errors

stakater-user commented:
@ahmadiq Yikes! You better fix it before anyone else finds out! [Build 9](https://jenkins.tools.stackator.com/job/Stakater/job/ProxyInjector/job/PR-1/9/) has Failed!

Commit:
fix import

stakater-user commented:
@ahmadiq Image is available for testing. ``docker pull stakater/proxyinjector:PR-1-10``

waseem-h commented on deployments/kubernetes/chart/proxyinjector/templates/configmap.yaml:
```diff
@@ -0,0 +1,37 @@
+    - {{ . }}
+    {{- end }}
+    resources:
+    {{- range $key, $value := index .resources 0}}
```
You should use toYaml in place of this

Commit:
support annotation for target port

stakater-user commented:
@ahmadiq Image is available for testing. ``docker pull stakater/proxyinjector:PR-1-11``

waseem-h commented on README.md:
```diff
@@ -1,2 +1,75 @@
+    | upstream-url  | url for the upstream endpoint you wish to proxy         |
+    | resources     | list of resources to proxy uri, methods, roles          |
+    | client-id     | client id used to authenticate to the oauth service     |
+    | client-secret | client secret used to authenticate to the oauth service |
```
What about the rest of the properties? Shouldn't they be documented? Or we can just link keycloak proxy's github repo

waseem-h commented on deployments/kubernetes/chart/proxyinjector/templates/configmap.yaml:
```diff
@@ -0,0 +1,37 @@
+data:
+  {{- with .Values.keycloakProxyConfigmap.config }}
+  config.yml: |-
+    client-id: {{ .clientId }}
```
We can just use toYaml here to pick up the whole config file from values.yaml
That way its up to the user to properly add config values and we won't have to manually add a new property whenever it comes in

waseem-h commented on internal/pkg/handler/create.go:
```diff
@@ -0,0 +1,139 @@
+	if r.Resource == nil {
+		logger.Errorf("Resource creation handler received nil resource")
+	} else {
+		name := callbacks.GetDeploymentName(r.Resource)
```
The Handle method is doing a lot of stuff. Refactor this into smaller and reusable methods.

waseem-h commented on pkg/kube/resourcemapper.go:
```diff
@@ -0,0 +1,11 @@
+
+// ResourceMap are resources from where changes are going to be detected
+var ResourceMap = map[string]runtime.Object{
+	"deployments": &v1beta1.Deployment{},
```
You're watching only deployments. Watch other resources too

ahmadiq replied:
getting some api related issues while handling daemonsets and statefulsets. Will resolve and handle in next version.

waseem-h commented on internal/pkg/handler/create.go:
```diff
@@ -0,0 +1,139 @@
+		logger.Errorf("Resource creation handler received nil resource")
+	} else {
+		name := callbacks.GetDeploymentName(r.Resource)
+		namespace := callbacks.GetDeploymentNamespace(r.Resource)
```
You're assuming that the resource can only be a deployment. It should cater all resources like daemonsets and statefulsets.

waseem-h commented on internal/pkg/handler/create.go:
```diff
@@ -0,0 +1,139 @@
+				}
+			}
+
+			payload := patch{
```
Refactor object creations to separate methods

waseem-h commented on internal/pkg/util/interface.go:
```diff
@@ -0,0 +1,38 @@
+package util
```
This whole file is unused. Please remove if not needed

Commit:
refactor

stakater-user commented:
@ahmadiq Image is available for testing. ``docker pull stakater/proxyinjector:PR-1-13``
